packages/wp-dashboard/src/api/hooks/useFontsApi.js

```diff
@@ -58,21 +60,30 @@ export default function useFontsApi() {
     async (id) => {
       try {
         const response = await deleteCustomFontCallback(fontsApiPath, id);
+        const newCustomFonts = customFonts.filter((font) => font.id !== id);
+        setCustomFonts(newCustomFonts);
         return response;
       } catch (e) {
         return null;
       }
     },
-    [fontsApiPath]
+    [customFonts, fontsApiPath]
   );
 
   const addCustomFont = useCallback(
-    (font) => addCustomFontCallback(fontsApiPath, font),
-    [fontsApiPath]
+    async (font) => {
+      const response = await addCustomFontCallback(fontsApiPath, font);
+      const { id, family, url } = response;
+      const newFont = { id, family, url };
+      const newCustomFonts = [newFont, ...customFonts];
+      setCustomFonts(newCustomFonts);
```

Comment:
Do we need the const here ` newCustomFonts` 

vs
 ```
 setCustomFonts([...customFonts, newFont]);
 ```

Reply:
New fonts should be added to the top of array.